File: javascript/examples/portfolio-monitor.ts
/**
 * Portfolio Monitor - Print a summary of all open positions.
 */

import { ProprClient } from '../propr-sdk';

const client = new ProprClient();
await client.setup();

const positions = await client.getOpenPositions();

let totalMargin = 0;
let totalUpnl = 0;

console.log('Asset        Side   Qty          Entry        Mark         uPnL');
console.log('-'.repeat(72));

for (const p of positions) {
  const margin = parseFloat(p.marginUsed);
  const upnl = parseFloat(p.unrealizedPnl);
  totalMargin += margin;
  totalUpnl += upnl;

  console.log(
    `${p.base.padEnd(12)} ${p.positionSide.padEnd(6)} ${p.quantity.padEnd(12)} ` +
    `${p.entryPrice.padEnd(12)} ${p.markPrice.padEnd(12)} ${upnl.toFixed(2)}`
  );
}

console.log('-'.repeat(72));
console.log(`Total margin: ${totalMargin.toFixed(2)} USDC`);
console.log(`Total uPnL:   ${totalUpnl.toFixed(2)} USDC`);

File: python/examples/portfolio_monitor.py
"""Portfolio Monitor - Print a summary of account and open positions."""

from decimal import Decimal
from propr_sdk import ProprClient

client = ProprClient()
client.setup()

# Get account state
account = client.get_account()
balance = Decimal(account["balance"])
available = Decimal(account["availableBalance"])
total_upnl = Decimal(account["totalUnrealizedPnl"])
equity = balance + total_upnl + Decimal(account["isolatedPositionMargin"])

print("Account Overview")
print("=" * 66)
print(f"Balance:       {balance:>12.2f} USDC")
print(f"Available:     {available:>12.2f} USDC")
print(f"uPnL:          {total_upnl:>12.2f} USDC")
print(f"Equity:        {equity:>12.2f} USDC")
print()

positions = client.get_open_positions()

if not positions:
    print("No open positions.")
else:
    print(f"{'Asset':<12} {'Side':<6} {'Qty':<12} {'Entry':<12} {'Mark':<12} {'uPnL':<12}")
    print("-" * 66)

    total_margin = Decimal("0")

    for p in positions:
        margin = Decimal(p["marginUsed"])
        upnl = Decimal(p["unrealizedPnl"])
        total_margin += margin

        print(f"{p['base']:<12} {p['positionSide']:<6} {p['quantity']:<12} "
              f"{p['entryPrice']:<12} {p['markPrice']:<12} {str(upnl):<12}")

    print("-" * 66)
    print(f"Total margin used: {total_margin:.2f} USDC")
